libs/inform/src/lib/services/modal/modal.service.ts

```diff
@@ -0,0 +1,234 @@
+import { Inject, Injectable, Optional, Type } from '@angular/core';
+import { Dialog } from '@angular/cdk/dialog';
+import {
+	combineLatest,
+	filter,
+	map,
+	NEVER,
+	Observable,
+	startWith,
+	Subject,
+	takeUntil,
+	tap,
+} from 'rxjs';
+
+import { NgxModalConfiguration, NgxModalOptions } from '../../types';
+import { NgxModalConfigurationToken } from '../../tokens';
+import { NgxModalAbstractComponent } from '../../abstracts';
+
+/**
+ * A wrapper service to Angular CDK Dialog that provides a WCAG/ARIA compliant implementation of modals
+ *
+ * @export
+ * @class NgxModalService
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxModalService {
+	/**
+	 * A subject to hold the close event
+	 */
+	private modalClosedSubject: Subject<void>;
+
+	constructor(
+		@Optional()
+		@Inject(NgxModalConfigurationToken)
+		private readonly configuration: NgxModalConfiguration,
+		private readonly dialogService: Dialog
+	) {}
+
+	/**
+	 * Opens a modal based on the provided options
+	 *
+	 * @param {NgxModalOptions<ActionType>} options - The modal options
+	 */
+	public open<ActionType extends string = string>(
+		options: NgxModalOptions<ActionType>
+	): Observable<ActionType> {
+		this.modalClosedSubject = new Subject();
+
+		// Iben: Fetch the type of component we wish to show
+		const configuration = this.configuration?.modals?.[options.type];
+		const component: Type<NgxModalAbstractComponent<ActionType>> =
+			options.component ||
+			(this.configuration?.modals?.[options.type].component as Type<
+				NgxModalAbstractComponent<ActionType>
+			>);
+
+		// Iben: Check if all the correct parameters are set and return NEVER when they're not correctly set
+		if (!this.runARIAChecks<ActionType>(options, component)) {
+			return NEVER;
+		}
+
+		// Iben: Render the modal
+		const modal = this.createModalComponent<ActionType>(options, component);
+
+		// Iben: Return the modal action
+		return combineLatest([
+			// Iben: Set the start value to undefined so both actions at least emit once
+			modal.action.pipe(startWith(undefined)),
+			modal.close.pipe(
+				// Iben: Map so we can keep the emit value void, but can work with the filter later down the line
+				map(() => 'NgxModalClose'),
+				// Iben: Set the start value to undefined so both actions at least emit once
+				startWith(undefined)
+			),
+		]).pipe(
+			// Iben: Only emit if one of the two actions actually has an emit
+			filter(([action, closed]) => {
```

Comment:
I would add interfaces in these arrow functions (filter, map, tap, ...) for clarity.